subsys/bluetooth/controller/ll_sw/ull.c

```diff
@@ -1370,10 +1370,37 @@ void ll_rx_mem_release(void **node_rx)
 		{
 			struct node_rx_sync *se =
 				(void *)((struct node_rx_pdu *)rx_free)->pdu;
+			uint8_t status = se->status;
+
+			/* Below status codes use node_rx_sync_estab, hence
+			 * release the node_rx memory.
+			 */
+			if ((status == BT_HCI_ERR_SUCCESS) ||
+			    (status == BT_HCI_ERR_UNSUPP_REMOTE_FEATURE) ||
+			    (status == BT_HCI_ERR_CONN_FAIL_TO_ESTAB)) {
+				struct ll_sync_set *sync;
+				struct ll_scan_set *scan;
+
+				scan = (void *)rx_free->rx_ftr.param;
 
-			if (!se->status) {
 				mem_release(rx_free, &mem_pdu_rx.free);
 
+				/* Sync established, do not release sync
+				 * context.
+				 */
+				if (status == BT_HCI_ERR_SUCCESS) {
+					ull_sync_setup_cleanup(scan);
+
+					break;
+				}
+
+				/* pick the sync context to be released */
+				sync = scan->per_scan.sync;
+
+				ull_sync_setup_cleanup(scan);
+
+				ull_sync_release(sync);
```

Comment:
```suggestion
                               /* Pick the sync context in case is has to be released */
				sync = scan->per_scan.sync;
				ull_sync_setup_cleanup(scan);

				if (status != BT_HCI_ERR_SUCCESS) {
        				ull_sync_release(sync);
				}
```